tables/tabulate-molecular-alterations.R

```diff
@@ -0,0 +1,264 @@
+# For further reference, see Figure 2:
+# https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/aa929753fca0019294571ec813e6ae7224b1d8b8/figures/pngs/figure2.png
+gene_list <- c("ACVR1", "APC", "ATM", "ATRX", "BCOR", "BRAF", 
+               "C11orf95", "CDK6", "CTDNEP1", "CTNNB1", "DDX3X", 
+               "EGFR", "ERBB4", "EWSR1", "FGFR1", "FGFR2", "FOXO3", 
+               "H3F3A", "IGF1R", "KDM6A", "KIAA1549", "KIT", 
+               "KMT2C", "KMT2D", "KRAS", "KSR2", "MAML2", "MAMLD1", 
+               "MET", "MIR512-2", "MYB", "MYCN", "NF1", "NF2", 
+               "NTRK2", "NTRK3", "PDGFRA", "PIK3CA", "PIK3R1", 
+               "PPM1D", "PRKAR1A", "PTCH1", "PTEN", "PTPN11", 
+               "QKI", "RAF1", "RELA", "ROS1", "SETD2", "SMARCA4", 
+               "TACC1", "TCF4", "TERT", "TP53", "YAP1", "ZIC1")
+
+select_cols <- c("Tumor_Sample_Barcode", "Hugo_Symbol", "Variant_Classification")
+
+
+# Combine all alterations:
+alterations_df <- list(maf_df, fusion_df, cnv_df) %>%
+  # keeping it interesting
+  purrr::map(
+    dplyr::select,
+    select_cols
+  ) %>%
+  dplyr::bind_rows() %>%
+  # rename Tumor_Sample_Barcode to sample_id
+  dplyr::rename(sample_id = Tumor_Sample_Barcode) %>%
+  # filter to only genes in gene_list
+  dplyr::filter(Hugo_Symbol %in% gene_list) %>%
+  # some wrangling to help with dev:
+  dplyr::arrange(Hugo_Symbol) %>%
+  tibble::as_tibble()
+
+# Ensure we still have all the genes at least _somewhere_ in the data frame
+if (!(length(unique(alterations_df$Hugo_Symbol)) == length(gene_list))) {
+  stop("Some genes are missing from the alteration df.")
+}
+
+# Convert to presence/absence of _all combinations_ of genes and variants for each sample_id
+alterations_presence_df <- alterations_df %>%
+  # If the row is here already, then the alteration is present
+  dplyr::mutate(present = "Y") %>% 
+  # Fill in all combinatins of gene/variant for each sample, 
+  #  filling in "N" (no) for present for any new rows 
+  tidyr::complete(sample_id, Hugo_Symbol, Variant_Classification, 
+                  fill = list(present = "N")) %>%
+  dplyr::distinct()
+
+  
+# At this point, alterations_presence_df is a Y/N tibble, but it contains only `sample_id`, not DNA and RNA biospecimen ids.
+# The final step is to add that information in, but we need to handle different groups separately:
+# - First, the fully non-ambiguous samples
+# - Second, the ambiguous samples: where DNA and RNA ids cannot be reliably linked, OR 
+#   where there are multiple same-modality experiments (since we are working with all samples, not `primary-only`)
+
+## Combine alterations_presence_df with biospecimen sample ids ----------------------
+  
+## First the fully non-ambiguous samples:
+export_identifiable_ids <- tumor_sample_ids_df %>%
+  # only ids of interest
+  dplyr::filter(!(ambiguous), 
+                !(multiple_exp_strategy)) %>%
```

Comment:
If we don't filter here, we will end up with multiple rows for each `sample_id` where there may be more than one DNA or RNA sample after joining. If we then pass this through a `group_by()`, we should be able to collapse DNA & RNA columns with a `paste` or similar, keeping everything else the same. 

Does that make sense as a strategy?

That at least should work for multiple modalities... I'm not sure what the ambiguous samples really look like.